### PR TITLE
Make DBI/SOS stackwalk work when thread is running native

### PR DESCRIPTION
**Pull request by Djuffin**
This change makes it possible for managed debuggers and SOS to obtain a managed stack for threads currently running native code or threads throwing managed exceptions.

**Works on Linux only**

### CODE REVIEW

**Comment by Djuffin** [comment]
cc @noahfalk

File: src/debug/di/shimdatatarget.h [hunk 1]
```diff
@@ -85,6 +82,13 @@ class ShimDataTarget : public ICorDebugMutableDataTarget
 
     // @dbgtodo - add Native Patch Table support
 
+    //
+    // ICorDebugDataTarget4
+    //    
+
+    // Get the OS Process ID that this DataTarget is for.
+    virtual DWORD STDMETHODCALLTYPE GetPid();
```
Comment (noahfalk) [inline src/debug/di/shimdatatarget.h]: Lets make this virtual HRESULT STDMETHODCALLTYPE GetPid(DWORD* pPid);

We want to be able to return errors like not implemented or not available without any confusion if a process ever has the same PID as one of those error constants.

File: src/inc/cordebug.idl [hunk 2]
```diff
@@ -810,6 +810,21 @@ interface ICorDebugDataTarget3 : IUnknown
...
+    object,
+    uuid(E799DC06-E099-4713-BDD9-906D3CC02CF2),
+    local,
+    pointer_default(unique)
+]
+interface ICorDebugDataTarget4 : IUnknown
+{
+    /*
+    * gives back a process id
+    */
```
Comment (noahfalk) [inline src/inc/cordebug.idl]: Can you add to this comment that the process could still be remote or a dump, and thus there is no guarantee that trying to use OS APIs against this PID will be fruitful.

**Comment by noahfalk** [comment]
LGTM, modulo those few comments above

**Pull request merged**